core/src/main/java/org/apache/spark/shuffle/sort/UnsafeShuffleWriter.java

```diff
@@ -281,25 +272,31 @@ void forceSorterToSpill() throws IOException {
    *
    * @return the partition lengths in the merged file.
    */
-  private long[] mergeSpills(SpillInfo[] spills, File outputFile) throws IOException {
+  private long[] mergeSpills(SpillInfo[] spills,
+      ShuffleMapOutputWriter mapWriter) throws IOException {
     final boolean compressionEnabled = (boolean) sparkConf.get(package$.MODULE$.SHUFFLE_COMPRESS());
     final CompressionCodec compressionCodec = CompressionCodec$.MODULE$.createCodec(sparkConf);
     final boolean fastMergeEnabled =
       (boolean) sparkConf.get(package$.MODULE$.SHUFFLE_UNDAFE_FAST_MERGE_ENABLE());
     final boolean fastMergeIsSupported = !compressionEnabled ||
       CompressionCodec$.MODULE$.supportsConcatenationOfSerializedStreams(compressionCodec);
     final boolean encryptionEnabled = blockManager.serializerManager().encryptionEnabled();
+    final int numPartitions = partitioner.numPartitions();
+    long[] partitionLengths = new long[numPartitions];
+    logger.error(mapWriter.toString());
     try {
       if (spills.length == 0) {
-        new FileOutputStream(outputFile).close(); // Create an empty file
-        return new long[partitioner.numPartitions()];
-      } else if (spills.length == 1) {
-        // Here, we don't need to perform any metrics updates because the bytes written to this
-        // output file would have already been counted as shuffle bytes written.
-        Files.move(spills[0].file, outputFile);
-        return spills[0].partitionLengths;
+        ShufflePartitionWriter writer = null;
+        try {
+          writer = mapWriter.getNextPartitionWriter();
+          writer.toStream();
```

Comment:
Do we need to do this?

Reply:
Without this the file does not get created and it fails `UnsafeShuffleWriter.writeEmptyIterator`

Reply:
When we commit all partitions, should we not check that we never created a writer; if we never did then we need to make the file?

Reply:
It seems that the test is expecting a file.... idk why, so maybe we should bring that file back?

Reply:
We've always expected a file, that's correct. The question is why do we have to call `writer.toStream` to ensure the file is created? I'm fairly certain the error is in how we wrote the plugin.

Reply:
Which is why, i think on `commit` we might need to create an empty file if a file does not exist. As we initially wrote but decided to take out

Reply:
We thought that `blockIndexResolver.writeIndexFileAndCommit` should create the file. Why is it not doing so in this case?